### DISCUSSION TITLE
Backfill assets on a partition schedule

### DISCUSSION THREAD

**Discussion by bmarcj** [dagster-io/dagster#18168] [Q&A] [answered]
I have a simple schedule that materialises a selection of daily partitioned assets:

```
build_schedule_from_partitioned_job(
    define_asset_job(
        name="assets",
        selection=AssetSelection.keys(...),
        partitions_def=PARTITION, 
       ...
```

For one (and only one) of these assets, I would like this schedule to include the materialisation of the last year of assets (i.e. if the schedule runs for 1st Jan 2023, materialising all other assets for that date, I would like one asset to backfill all its 2022 partitions back to 1st Jan 2022, as well as 1st Jan 2023).

What's the best way to do this?

**Comment by clairelin135** [comment, marked as answer]
Runs currently must target the same partition(s) for each asset in the job, so in your schedule you'll need to return additional run requests that target the remainder of the partitions. 

This would require defining a custom schedule, which allows you to return custom run requests. [Example here. ](https://github.com/dagster-io/dagster/discussions/15532)

In order to build run requests that target all partitions of the special asset, you could either:
(1) Yield a run request per partition
(2) [Yield a run request that targets a range of partitions](https://github.com/dagster-io/dagster/discussions/14622)
